pkg/sync/adapter.go

```diff
@@ -0,0 +1,226 @@
+package sync
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/exec"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/util"
+
+	"github.com/pkg/errors"
+)
+
+// New instantiantes a component adapter
+func New(adapterContext common.AdapterContext, client SyncClient) Adapter {
+	return Adapter{
+		Client:         client,
+		AdapterContext: adapterContext,
+	}
+}
+
+// Adapter is a component adapter implementation for sync
+type Adapter struct {
+	Client SyncClient
+	common.AdapterContext
+}
+
+// SyncFiles does a couple of things:
+// if files changed/deleted are passed in from watch, it syncs them to the component
+// otherwise, it checks which files have changed and syncs the delta
+// it returns a boolean execRequired and an error. execRequired tells us if files have
+// changed and devfile execution is required
+func (a Adapter) SyncFiles(syncParameters common.SyncParameters) (isPushRequired bool, err error) {
+
+	deletedFiles := []string{}
+	changedFiles := []string{}
+	isForcePush := false
+	pushParameters := syncParameters.PushParams
+	compInfo := syncParameters.CompInfo
+	globExps := util.GetAbsGlobExps(pushParameters.Path, pushParameters.IgnoredFiles)
```

Comment:
That's a lot of above variables, maybe put it in a struct instead? Up to you, just a suggestion.

Reply:
cleanup tracking with https://github.com/openshift/odo/issues/3015

Reply:
clean up tracked with https://github.com/openshift/odo/issues/3015